localstack/ext/java/src/test/java/cloud/localstack/BasicFunctionalityTest.java

```diff
@@ -113,6 +117,40 @@ public void testKinesisLambdaIntegration() throws Exception {
         // TODO: have Lambda store the record to S3, retrieve it from there, compare result
     }
 
+    @org.junit.Test
+    @org.junit.jupiter.api.Test
+    public void testSQSLambdaIntegration() throws Exception {
+        AmazonSQS clientSQS = TestUtils.getClientSQS();
+        AWSLambda lambda = TestUtils.getClientLambda();
+        String functionName = UUID.randomUUID().toString();
+        String sqsQueueName = UUID.randomUUID().toString();
+
+        // create function
+        CreateFunctionRequest request = new CreateFunctionRequest();
+        request.setFunctionName(functionName);
+        request.setRuntime(Runtime.Java8);
+        request.setCode(LocalTestUtil.createFunctionCode(SQSLambdaHandler.class));
+        request.setHandler(SQSLambdaHandler.class.getName());
+        lambda.createFunction(request);
+
+        // create stream
+        CreateQueueResult queue = clientSQS.createQueue(sqsQueueName);
+        Thread.sleep(500);
+        GetQueueAttributesResult queueAttributes = clientSQS.getQueueAttributes(new GetQueueAttributesRequest()
+                .withQueueUrl(queue.getQueueUrl())
+                .withAttributeNames(QueueAttributeName.QueueArn));
+        String streamArn = queueAttributes.getAttributes().get(QueueAttributeName.QueueArn.name());
+
+        // create mapping
+        CreateEventSourceMappingRequest mapping = new CreateEventSourceMappingRequest();
+        mapping.setFunctionName(functionName);
+        mapping.setEventSourceArn(streamArn);
+        lambda.createEventSourceMapping(mapping);
+
+        // push event
+        clientSQS.sendMessage(queue.getQueueUrl(), "{\"foo\": \"bar\"}");
+    }
```

Comment:
Wondering if we could add a simple assertion here to make sure the Lambda was actually triggered from the SQS message. For example, we could have the Lambda put a dummy file to an S3 bucket, and then add an assertion to this test which checks that the S3 object actually got created. Do you think we could add such a simple check to this PR @Psynbiotik ?

Reply:
@whummer Good idea, I'll add the test.

Reply:
I've added the requested test.

It was quite the challenge, I found one bug in the python code and added quite a few fixes for the way the java tests were being run.

Now java tests will actually use the latest code instead of pulling from the git repo, this applies to both java code and python code, previously any changes you made to non-test code would not be used when the tests ran, instead using code pulled from the git repo.